tasks/eth/013-fp-upgrade-fjord/VALIDATION.md

```diff
@@ -0,0 +1,91 @@
+  **Value:** 0x0000000000000000000000000000000000000000000000000000000000000001 <br/>
+  **Meaning:** This is `owners[0xca11bde05977b3631167028862be2a173976ca11] -> 1`, so the key can be
+    derived from `cast index address 0xca11bde05977b3631167028862be2a173976ca11 2`.
+
+- **Key:** 0xe90b7bceb6e7df5418fb78d8ee546e97c83a08bbccc01a0644d599ccd2a7c2e0 <br/>
+  **Value:** 0x000000000000000000000000ca11bde05977b3631167028862be2a173976ca11 <br/>
+  **Meaning:** This is `owners[1] -> 0xca11bde05977b3631167028862be2a173976ca11`, so the key can be
+    derived from `cast index address 0x0000000000000000000000000000000000000001 2`.
+
+## State Changes
+
+### `0xe5965ab5962edc7477c8520243a95517cd252fa9` (`DisputeGameFactoryProxy`)
+
+Links:
+- [Etherscan](https://etherscan.io/address/0xe5965ab5962edc7477c8520243a95517cd252fa9)
+- [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/extra/addresses/mainnet/op.json)
+
+State Changes:
+**Note:** The `101` referenced below is the storage slot of `gameImpls` defined in the [DisputeGameFactory storage layout](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.4.0/packages/contracts-bedrock/snapshots/storageLayout/DisputeGameFactory.json#L37C1-L43C5).
+
+- **Key:** `0x4d5a9bd2e41301728d41c8e705190becb4e74abe869f75bdb405b63716a35f9e` <br/>
+  **Before:** `0x000000000000000000000000e9dad167ef4de8812c1abd013ac9570c616599a0` <br/>
+  **After:** [`0x000000000000000000000000c307e93a7c530a184c98eade4545a412b857b62f`](https://etherscan.io/address/0xc307e93a7c530a184c98eade4545a412b857b62f) <br/>
+  **Meaning:** This is `gameImpls[0] -> 0xc307e93a7c530a184c98eade4545a412b857b62f` (where `0` is the [`CANNON` game type](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.4.0/packages/contracts-bedrock/src/dispute/lib/Types.sol#L28), so the key can be derived from `cast index uint32 0 101`.
+
+- **Key:** `0xffdfc1249c027f9191656349feb0761381bb32c9f557e01f419fd08754bf5a1b` <br/>
+  **Before:** `0x0000000000000000000000004146df64d83acb0dcb0c1a4884a16f090165e122` <br/>
+  **After:** [`0x000000000000000000000000f691f8a6d908b58c534b624cf16495b491e633ba`](https://etherscan.io/address/0xf691f8a6d908b58c534b624cf16495b491e633ba) <br/>
+  **Meaning:** This is `gameImpls[1] -> 0xf691f8a6d908b58c534b624cf16495b491e633ba` (where `1` is the [`PERMISSIONED_CANNON` game type](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.4.0/packages/contracts-bedrock/src/dispute/lib/Types.sol#L31)), so the key can be derived from `cast index uint32 0 101`.
+
+### Safe Contract State Changes
+
+The only other state changes should be restricted to one of the following addresses:
+
+- L1 2/2 ProxyAdmin Owner Safe: `0x5a0Aae59D09fccBdDb6C6CcEB07B7279367C3d2A`
+  - The nonce (slot 0x5) should be increased from 4 to 5.
+  - Another key is set from 0 to 1 reflecting an entry in the `approvedHashes` mapping.
```

Comment:
We need to explain how to verify that this key is correct. Let's pull this out into a ful `### 0x5a0Aae59D09fccBdDb6C6CcEB07B7279367C3d2A` state changes entry. Here is a reference implementation, you will just need to change the keys and the safe hashes: https://github.com/ethereum-optimism/superchain-ops/blob/main/tasks/eth/010-1-guardian-upgrade/VALIDATION.md#0x5a0aae59d09fccbddb6c6cceb07b7279367c3d2a-the-22-proxyadmin-owner-1

Reply:
That example was very useful. Do we also want to specify the expected nonce changes for the SC and FND safes? Or is that redundant given that the approval hash includes it?

Reply:
I think it's good to specify, it gives extra confidence to signers that they are signing the right thing, since it's more user-friendly than the approveHash